src/protocol/attribution/credit_capping.rs

```diff
@@ -73,8 +74,7 @@ where
         .enumerate()
         .map(|(i, x)| {
             MCCreditCappingOutputRow::new(x.breakdown_key.clone(), final_credits[i].clone())
-        })
-        .collect::<Vec<_>>();
+        });
```

Comment:
This does not work. Error  at line 79
note: expected opaque type `impl Iterator<Item = MCCreditCappingOutputRow<F, T>> + '_`
                    found struct `std::iter::Map<std::iter::Enumerate<std::slice::Iter<'_, MCAccumulateCreditInputRow<F, T>>>, [closure@src/protocol/attribution/credit_capping.rs:75:14: 75:22]>`

Reply:
I think the problem is actually in line 31. This function attempts to return two different iterator types : one from credit_capping_max_one which returns `impl Iterator` and another one from line 79 (that is a different `impl Iterator`).  If you comment lines 30-32 this will compile. Rust compiler forces functions to have only one return type and that is what is preventing this code from compile. 

How to avoid this is a hard problem though. If you want to use `impl Iterator` in return type, there must be only one return point from this function. Alternatively, it could be possible to create a struct that implements `Iterator<Item = MCCreditCappingOutputRow>` and return it instead - returning a struct is no longer confusing to Rust compiler